### PR TITLE
feat(component): listoptimizer component

### PR DESCRIPTION
**Pull request by mukiienko**

### CODE REVIEW

File: src/components/List/ListOptimizer/ListOptimizer.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import { List, CellMeasurer, CellMeasurerCache, AutoSizer, ListRowProps } from 'react-virtualized';
+
+export interface Props {
+    rowCount: number;
+    defaultHeight?: number;
+    defaultWidth?: number;
+    children: (props: ListRowProps) => React.ReactNode;
+}
+
+const DEFAULT_HEIGHT = 30;
```
Comment (igorarkhipenko) [inline src/components/List/ListOptimizer/ListOptimizer.tsx]: maybe it makes sense to rename this constant to `DEFAULT_CELL_HEIGHT` or something? Current naming in combination with `defaultHeight` prop of the component below makes it a bit ambiguous.

Comment (mukiienko) [reply src/components/List/ListOptimizer/ListOptimizer.tsx]: Yep, good point

Comment (mukiienko) [reply src/components/List/ListOptimizer/ListOptimizer.tsx]: Good point

File: src/components/Select/SuggestionsList/SuggestionsList.tsx [hunk 2]
```diff
@@ -36,26 +39,42 @@ export function SuggestionsList<S>(props: Props<S>) {
...
-                    key={suggestionToString(item)}
-                    {...getItemProps({
-                        item,
-                        index,
+            {useOptimizeRender ? (
+                <ListOptimizer rowCount={suggestions.length}>{renderItem}</ListOptimizer>
+            ) : (
+                <>
+                    {suggestions.map((item, index) => {
+                        const key = suggestionToString(suggestions[index]);
```
Comment (igorarkhipenko) [inline src/components/Select/SuggestionsList/SuggestionsList.tsx]: can be replaced with `const key = suggestionToString(item);`

**Pull request merged**